src/AutoMapper/Bundle/AutoMapper.php

```diff
@@ -3,25 +3,13 @@
 namespace Jane\AutoMapper\Bundle;
 
 use Jane\AutoMapper\AutoMapper as BaseAutoMapper;
-use Jane\AutoMapper\Bundle\Configuration\ConfigurationPassInterface;
-use Jane\AutoMapper\MapperGeneratorMetadataInterface;
+use Jane\AutoMapper\Bundle\Configuration\MapperConfigurationInterface;
 
 class AutoMapper extends BaseAutoMapper
 {
-    /** @var ConfigurationPassInterface[] */
-    private $configurationPass = [];
-
-    public function addConfigurationPass(ConfigurationPassInterface $configurationPass): void
+    public function addMapperConfiguration(MapperConfigurationInterface $mapperConfiguration): void
     {
-        $this->configurationPass[] = $configurationPass;
-    }
-
-    public function register(MapperGeneratorMetadataInterface $metadata): void
-    {
-        foreach ($this->configurationPass as $configurationPass) {
-            $configurationPass->process($metadata);
-        }
-
-        parent::register($metadata);
+        $metadata = $this->getMetadata($mapperConfiguration->getSource(), $mapperConfiguration->getTarget());
+        $mapperConfiguration->process($metadata);
     }
```

Comment:
Another issue here: When the MapperConfiguration will be registered in the AutoMapper now if you remove this code ?

Reply:
The `MapperConfiguration` goal is to alter a metadata. Once done, we don't need to keep track of the `MapperConfiguration` instance.

Reply:
Sure but, how the AutoMapper knows about theses configurations ?

Reply:
Calling the `getMetadata` method creates & register (if needed) a `MapperGeneratorMetadataInterface` instance.
This instance is then processed by the `MapperConfigurationInterface`, that's why we don't need to do anything else.

I thought it was for this reason you suggested to remove the call to `parent::register($metadata)`.

Reply:
Ok, I was missing this `getMetadata` call :+1: